src/main/java/com/mongodb/hibernate/dialect/MongoDialectSettings.java

```diff
@@ -0,0 +1,293 @@
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.service.spi.Configurable;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * The configuration of {@link MongoDialect}, {@link MongoConnectionProvider}, the internal {@link MongoClient}.
+ *
+ * <table>
+ *     <caption>Supported configuration properties</caption>
+ *     <thead>
+ *         <tr>
+ *             <th>Method</th>
+ *             <th>Has default</th>
+ *             <th>Related {@linkplain Configurable#configure(Map) configuration property} name</th>
+ *             <th>Supported value types of the configuration property</th>
+ *             <th>Value, unless overridden via {@link Builder}</th>
+ *         </tr>
+ *     </thead>
+ *     <tbody>
+ *         <tr>
+ *             <td>{@link #getMongoClientSettings()}</td>
+ *             <td>✓</td>
+ *             <td>{@value AvailableSettings#JAKARTA_JDBC_URL}</td>
+ *             <td>
+ *                 <ul>
+ *                     <li>{@link String}</li>
+ *                     <li>{@link ConnectionString}</li>
+ *                 </ul>
+ *             </td>
+ *             <td>Is {@linkplain MongoClientSettings.Builder#applyConnectionString(ConnectionString) based} on
+ *             the {@link ConnectionString} {@linkplain ConnectionString#ConnectionString(String) constructed} from
+ *             {@value AvailableSettings#JAKARTA_JDBC_URL}, if the latter is configured; otherwise a {@link MongoClientSettings}
+ *             instance with its defaults.</td>
+ *         </tr>
+ *         <tr>
+ *             <td>{@link #getDatabaseName()}</td>
+ *             <td>✗</td>
+ *             <td>{@value AvailableSettings#JAKARTA_JDBC_URL}</td>
+ *             <td>
+ *                 <ul>
+ *                     <li>{@link String}</li>
+ *                     <li>{@link ConnectionString}</li>
+ *                 </ul>
+ *             </td>
+ *             <td>The MongoDB database name from {@value AvailableSettings#JAKARTA_JDBC_URL},
+ *             if {@linkplain ConnectionString#getDatabase() configured};
+ *             otherwise there is no default value, and a value must be configured via {@link Builder#databaseName(String)}.</td>
+ *         </tr>
+ *         <tr>
+ *             <td>{@link #isAutoCommit()}</td>
+ *             <td>✓</td>
+ *             <td>{@value AvailableSettings#AUTOCOMMIT}</td>
+ *             <td>
+ *                 <ul>
+ *                     <li>{@link String}, only {@code "true"}, {@code "false"} are allowed</li>
+ *                     <li>{@link Boolean}</li>
+ *                 </ul>
+ *             </td>
+ *             <td>{@value AvailableSettings#AUTOCOMMIT}, if configured; otherwise {@code false}.</td>
+ *         </tr>
+ *     </tbody>
+ * </table>
+ *
+ * @see MongoDialectConfigurator
+ */
+public final class MongoDialectSettings {
+    private final MongoClientSettings mongoClientSettings;
+    private final String databaseName;
+    private final boolean autoCommit;
```

Comment:
We may trivially support configuring an external `MongoClient` via `MongoDialectSettings`, as opposed to creating an internal one, enalbing applications to reuse clients.

Reply:
I am not sure `autoCommit` is a good candidate for it seems an open question now. In JDBC, regardless of whether commit is auto, commit happens all the time. Currently we only commit in face of `autoCommit` value is false.

Reply:
I doubt whether we need to include autoCommit() for it is internal Hibernate implememntatoin detail and its business logic might be complex that "we set a Connection's autoCommit status after it was created". That part might interfere with Hibernate's internal working. For instance, without our doing anything, the Connection would be set autoCommit as expected.

Reply:
I agree that auto-commit shouldn't be configurable programmatically via our API.

Done in 0155ba09610e16853c3e6c9fbf41302186c39b8d.